src/details/ArborX_DetailsUtils.hpp

```diff
@@ -148,47 +170,72 @@ void iota(Kokkos::View<T, P...> const &v,
+  Kokkos::parallel_for("iota", policy,
                        KOKKOS_LAMBDA(int i) { v(i) = value + (ValueType)i; });
 }
 
+template <typename T, typename... P>
+[[deprecated]] inline void
+iota(Kokkos::View<T, P...> const &v,
+     typename Kokkos::ViewTraits<T, P...>::value_type value = 0)
+{
+  using ExecutionSpace = typename Kokkos::ViewTraits<T, P...>::execution_space;
+  iota(ExecutionSpace{}, v, value);
+}
+
 /** \brief Returns the smallest and the greatest element in the view
  *
+ *  \param[in] space Execution space
  *  \param[in] v Input view
  *
  *  Returns a pair on the host with the smallest value in the view as the first
  *  element and the greatest as the second.
  */
-template <typename ViewType>
+template <typename ExecutionSpace, typename ViewType>
 std::pair<typename ViewType::non_const_value_type,
           typename ViewType::non_const_value_type>
-minMax(ViewType const &v)
+minMax(ExecutionSpace &&space, ViewType const &v)
 {
   static_assert(ViewType::rank == 1, "minMax requires a View of rank 1");
-  using ExecutionSpace = typename ViewType::execution_space;
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
   Kokkos::MinMaxScalar<typename ViewType::non_const_value_type> result;
   Kokkos::MinMax<typename ViewType::non_const_value_type> reducer(result);
-  Kokkos::parallel_reduce("minMax", Kokkos::RangePolicy<ExecutionSpace>(0, n),
+  Kokkos::RangePolicy<std::remove_reference_t<ExecutionSpace>> policy(
+      std::forward<ExecutionSpace>(space), 0, n);
+  Kokkos::parallel_reduce("minMax", policy,
                           Kokkos::Impl::min_max_functor<ViewType>(v), reducer);
   return std::make_pair(result.min_val, result.max_val);
 }
 
+template <typename ViewType>
+[[deprecated]] inline std::pair<typename ViewType::non_const_value_type,
+                                typename ViewType::non_const_value_type>
+minMax(ViewType const &v)
+{
+  using ExecutionSpace = typename ViewType::execution_space;
+  return minMax(ExecutionSpace{}, v);
+}
+
 /** \brief Returns the smallest element in the view
  *
+ *  \param[in] space Execution space
  *  \param[in] v Input view
  */
-template <typename ViewType>
-typename ViewType::non_const_value_type min(ViewType const &v)
+template <typename ExecutionSpace, typename ViewType>
+typename ViewType::non_const_value_type min(ExecutionSpace &&space,
+                                            ViewType const &v)
 {
   static_assert(ViewType::rank == 1, "min requires a View of rank 1");
-  using ExecutionSpace = typename ViewType::execution_space;
   auto const n = v.extent(0);
   ARBORX_ASSERT(n > 0);
   typename ViewType::non_const_value_type result;
   Kokkos::Min<typename ViewType::non_const_value_type> reducer(result);
-  Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecutionSpace>(0, n),
+  Kokkos::RangePolicy<std::remove_reference_t<ExecutionSpace>> policy(
+      std::forward<ExecutionSpace>(space), 0, n);
+  Kokkos::parallel_reduce("min", policy,
```

Comment:
Nice catch with the "min".

Reply:
Should probably be `ARBORX_MARK_REGION`.

Reply:
Done